### PR TITLE
fix: Update Docker build to automatically adapt Torch version

### PR DESCRIPTION
**Pull request by gs-olive**
# Description

- Build issue encountered in Docker arising from mismatched versions in WORKSPACE and Docker folder
- Update Dockerfile build process to adapt Torch version from `requirements.txt`
- Remove references to Torch version in Docker `WORKSPACE` file; instead use local Torch version

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [ x ] My code follows the style guidelines of this project (You can use the linters)
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ x ] I have made corresponding changes to the documentation
- [ x ] I have added tests to verify my fix or my feature
- [ x ] New and existing unit tests pass locally with my changes
- [ x ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -13,7 +13,7 @@ RUN python get-pip.py
 RUN pip3 install wheel
 
 # Install Pytorch
-RUN pip3 install torch==2.0.0.dev20230103+cu117 torchvision==0.15.0.dev20230103+cu117 --extra-index-url https://download.pytorch.org/whl/nightly/cu117
+RUN pip3 install torch==2.0.0.dev20230219+cu117 torchvision==0.15.0.dev20230219+cu117 --extra-index-url https://download.pytorch.org/whl/nightly/cu117
```
Comment (narendasan) [inline docker/Dockerfile:16]: Can we just reference the `requirements.txt` so this doesnt need to be manually updated?

File: docker/WORKSPACE.docker [hunk 2]
```diff
@@ -51,17 +51,17 @@ new_local_repository(
...
-    urls = ["https://download.pytorch.org/libtorch/nightly/cu117/libtorch-cxx11-abi-shared-with-deps-2.0.0.dev20230103%2Bcu117.zip"],
+    urls = ["https://download.pytorch.org/libtorch/nightly/cu117/libtorch-cxx11-abi-shared-with-deps-2.0.0.dev20230219%2Bcu117.zip"],
 )
 
 http_archive(
     name = "libtorch_pre_cxx11_abi",
     build_file = "@//third_party/libtorch:BUILD",
-    sha256 = "e260fc7476be89d1650953e8643e9f7363845f5a52de4bab87ac0e619c1f6ad4",
+    sha256 = "aa7fd06079d260ff83c344d043fb84fbd9cf831cf375ed8b5a1b62416817af31",
     strip_prefix = "libtorch",
```
Comment (narendasan) [inline docker/WORKSPACE.docker:63]: ideas on if we can avoid needing http archive here?

**Review by narendasan — APPROVED** [review]

File: docker/Dockerfile (hunk 1 above)
Comment (gs-olive) [reply docker/Dockerfile]: Used reference to `requirements.txt` to avoid explicit version naming here.

File: docker/WORKSPACE.docker (hunk 2 above)
Comment (gs-olive) [reply docker/WORKSPACE.docker]: Used local version of Torch in container to avoid use of `http_archive`.

File: docker/Dockerfile
Comment (narendasan) [inline docker/Dockerfile:1]: Add an `ARG` called `USE_CXX11_ABI` and add a command to the dockerfile like 

```Dockerfile
ARG USE_CXX11_ABI
ENV USE_CXX11=USE_CXX11_ABI

```

File: docker/README.md [hunk 3]
```diff
@@ -5,7 +5,8 @@
 * `Dockerfile` currently uses the exact library versions (Torch, CUDA, CUDNN, TensorRT) listed in <a href="https://github.com/pytorch/TensorRT#dependencies">dependencies</a> to build Torch-TensorRT.
 
 * This `Dockerfile` installs `pre-cxx11-abi` versions of Pytorch and builds Torch-TRT using `pre-cxx11-abi` libtorch as well.
-Note: To install `cxx11_abi` version of Torch-TensorRT, enable `USE_CXX11=1` flag so that `dist-build.sh` can build it accordingly.
+
+Note: Torch-TRT based on the C++11 ABI is not automatically installed in the build. To install it, modify the `WORKSPACE.docker` file to specify a C++11 ABI path, then enable the `USE_CXX11=1` flag so that `dist-build.sh` can build it accordingly. See the `WORKSPACE` file in the root of the repository for an example.
```
Comment (narendasan) [inline docker/README.md:9]: Note: By default the container uses the pre_cxx11_abi. If you are using a workflow that requires a build of PyTorch on the CXX11_ABI (e.g. using the PyTorch NGC containers as a base image), add the build-arg `--build-arg USE_CXX11_ABI=1`

File: docker/Dockerfile
Comment (gs-olive) [inline docker/Dockerfile:1]: Remove `pip install tensorrt` direct call

Comment (narendasan) [inline docker/Dockerfile:1]: use 22.04 base

**Comment by gs-olive** [comment]
22.04 base upgrade and tensorrt/cuddn build args to be addressed in Issue #1782

**Pull request merged**